gyronimo/core/contraction.cc

```diff
@@ -153,6 +153,111 @@ IR3 contraction<second, third>(const ddIR3& ddA, const IR3& B, const IR3& C) {
   };
 }
 
+//! One index contraction of `SM3` object and First index of `ddIR3` object.
+/*!
+    Returns the object
+    ```
+    ddC[ddIR3::ijk] = g[SM3::iu] * ddA[ddIR3::ujk] +
+			+ g[SM3::iv] * ddA[ddIR3::vjk] + g[SM3::iw] * ddA[ddIR3::wjk]
+    ```
+    with `i,j,k = u, v, w`.
+    Replacements SM3::ij -> SM3::ji and ddIR3::ikj -> ddIR3::ijk have been
+    explicitly performed.
+*/
+template<>
+ddIR3 contraction<second, first>(const SM3& g, const ddIR3& ddA) {
```

Comment:
Wouldn't this definition be equivalent to `ddIR3 contraction<first>(const ddIR3& ddA, const SM3& g)`? Because `SM3` are symmetric, we don't need to specify its contracted index. Is this assumption correct? Does it mess with other parts of the code?

Reply:
I believe your assumption is correct. The reason I coded it that way is that even though SM3 is symmetric, it is still a matrix - hence the specification of the contraction index. This expression is used in the conversion between chrystoffel symbols of first and second kind and may be used someplace else, but it's no burden to change it. Should I switch it to `ddIR3 contraction<first>(...)`?